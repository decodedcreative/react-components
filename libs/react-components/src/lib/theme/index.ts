import { DefaultTheme } from 'styled-components';

const lightTheme: DefaultTheme = {
  borderRadius: '5px',
  colors: {
    primary: '#00aeef',
    interactive: '#006de3',
    interactiveAlt: '#005bbe',
    error: '#e90014',
    errorTint: '#f9e6e7',
    warning: '#ffba00',
    warningTint: '#fef8e7',
    success: '#158724',
    successTint: '#eaf8e7',
    color0: '#ffffff',
    color10: '#ffffff',
    color20: '#f2f7ff',
    color30: '#f6f6f6',
    color40: '#e2e2e2',
    color50: '#8e8e8e',
  },
  button: {
    padding: '10px 28px',
    borderWidth: '2px',
  },
};

export { lightTheme };
