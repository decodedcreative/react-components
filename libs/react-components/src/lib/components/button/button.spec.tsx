import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../../theme/';
import Button, { ButtonProps } from './button';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

const createComponentJSX = ({
  children = 'Button text',
  variant,
  disabled,
  type = 'button',
}: ButtonProps) => (
  <ThemeProvider theme={lightTheme}>
    <Button variant={variant} type={type} disabled={disabled}>
      {children}
    </Button>
  </ThemeProvider>
);

const createComponentRender = ({
  children = 'Button text',
  variant,
  disabled,
  type = 'button',
}: ButtonProps) =>
  render(
    <ThemeProvider theme={lightTheme}>
      <Button variant={variant} type={type} disabled={disabled}>
        {children}
      </Button>
    </ThemeProvider>
  );

describe('Button', () => {
  it('should render successfully', () => {
    const component = createComponentRender({} as ButtonProps);
    expect(component).toBeTruthy();
  });

  describe('Button - Variants', () => {
    it('should match the secondary button snapshot', () => {
      const secondaryButton = createComponentJSX({} as ButtonProps);
      const secondaryButtonJSON = renderer.create(secondaryButton).toJSON();
      expect(secondaryButtonJSON).toMatchSnapshot();
    });
    it('should match the primary button snapshot', () => {
      const primaryButton = createComponentJSX({
        variant: 'primary',
      } as ButtonProps);
      const primaryButtonJSON = renderer.create(primaryButton).toJSON();
      expect(primaryButtonJSON).toMatchSnapshot();
    });
  });

  describe('Button - Disabled', () => {
    it('should become disabled if a disabled prop is supplied', () => {
      const component = createComponentRender({
        children: 'Disabled button',
        disabled: true,
      } as ButtonProps);
      const button = component.getByText(/Disabled button/i);
      expect(button).toHaveAttribute('disabled');
    });
    it('should match the disabled button snapshot', () => {
      const disabledButton = createComponentJSX({
        children: 'Disabled button',
        disabled: true,
      } as ButtonProps);
      const disabledButtonJSON = renderer.create(disabledButton).toJSON();
      expect(disabledButtonJSON).toMatchSnapshot();
    });
  });
});
