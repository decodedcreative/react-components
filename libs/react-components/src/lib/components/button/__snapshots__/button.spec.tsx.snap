// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button - Disabled should match the disabled button snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  border: 2px solid #006de3;
  color: #006de3;
  cursor: pointer;
  word-break: break-word;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 5px;
  padding: 10px 28px;
}

.c0:disabled,
.c0:disabled:hover,
.c0[aria-disabled='true'],.c0[aria-disabled='true']:hover {
  background-color: #f6f6f6;
  color: #8e8e8e;
  border-color: #8e8e8e;
}

<button
  className="c0"
  disabled={true}
  type="button"
>
  Disabled button
</button>
`;

exports[`Button Button - Variants should match the primary button snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  border: 2px solid #006de3;
  color: #006de3;
  cursor: pointer;
  word-break: break-word;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 5px;
  padding: 10px 28px;
  background-color: #006de3;
  color: #ffffff;
}

.c0:disabled,
.c0:disabled:hover,
.c0[aria-disabled='true'],.c0[aria-disabled='true']:hover {
  background-color: #f6f6f6;
  color: #8e8e8e;
  border-color: #8e8e8e;
}

<button
  className="c0"
  type="button"
>
  Button text
</button>
`;

exports[`Button Button - Variants should match the secondary button snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  border: 2px solid #006de3;
  color: #006de3;
  cursor: pointer;
  word-break: break-word;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 5px;
  padding: 10px 28px;
}

.c0:disabled,
.c0:disabled:hover,
.c0[aria-disabled='true'],.c0[aria-disabled='true']:hover {
  background-color: #f6f6f6;
  color: #8e8e8e;
  border-color: #8e8e8e;
}

<button
  className="c0"
  type="button"
>
  Button text
</button>
`;
